src/controllers/blogs-controllers.js

```diff
@@ -0,0 +1,95 @@
+import blogs from '../models/blogs-collection';
+import { v4 as uuidv4 } from 'uuid';
+
+
+export const readAll = (req, res) => {
+    if (!blogs) {
+        return res.status(404).json({
+            status: 404,
+            error: 'Blog not accessible',
+        });
+    }
+    return res.status(200).json({
+        status: 200,
+        blog: 'blogs successfully retrieved',
+        data: {
+            blogs,
+        },
+    });
+};
+export const readById = (req, res) => {
+    const id = req.params.id;
+    const blog = blogs.filter((user) => {
+        return user.id === id;
+    });
+    if (blog[0]) {
+        return res.status(200).json({
+            status: 200,
+            blog: 'blog successfully retrieved',
+            data: blog,
+        });
+    }
+    return res.status(404).json({
+        status: 404,
+        error: 'blog can not be found',
+    });
+
+};
+export const create = (req, res) => {
+    const blog = {
+        id: uuidv4(),
+        author:req.body.author,
+        title: req.body.title,
+        description:req.body.description,
+    };
+    blogs.push(blog);
+    return res.status(201).json({
+        status: 201,
+        blog: 'blog successfully created',
+        data: blog,
+    });
+
+};
+
+export const update=(req,res)=>{
+    const id = req.params.id;
+    const blog = blogs.filter((blog) => {
+        return blog.id === id;
```

Comment:
You should use **Find**  not `filter`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find